ps_crossselling.php

```
@@ -148,200 +216,198 @@ public function renderForm()
-            $presenter = new ProductListingPresenter(
-                new ImageRetriever(
-                    $this->context->link
-                ),
-                $this->context->link,
-                new PriceFormatter(),
-                new ProductColorsRetriever(),
-                $this->context->getTranslator()
-            );
+        return $orderedProductIds;
+    }
 
-            $productsForTemplate = [];
+    /**
+     * Clear cache after product change
+     */
+    public function hookActionObjectProductUpdateAfter()
+    {
+        $this->clearCache();
+    }
 
-            $presentationSettings->showPrices = $showPrice;
+    /**
+     * Clear cache after product deletion
+     */
+    public function hookActionObjectProductDeleteAfter()
+    {
+        $this->clearCache();
+    }
 
-            if (is_array($order_products)) {
-                foreach ($order_products as $productId) {
-                    $productsForTemplate[] = $presenter->present(
-                        $presentationSettings,
-                        $assembler->assembleProduct(['id_product' => $productId['product_id']]),
-                        $this->context->language
-                    );
-                }
-            }
+    /**
+     * @return bool
+     */
+    protected function clearCache()
+    {
+        // Avoid cache clearing on PrestaShop installation
+        if (defined('PS_INSTALLATION_IN_PROGRESS') && constant('PS_INSTALLATION_IN_PROGRESS')) {
+            return true;
+        }
 
-            return $productsForTemplate;
+        // Avoid cache clearing on product import
+        if (defined('PS_MASS_PRODUCT_CREATION') && constant('PS_MASS_PRODUCT_CREATION')) {
+            return true;
         }
 
-        return false;
+        try {
+            /** @var FilesystemCache $cache */
+            $cache = $this->get('ps_crossselling.cache.products');
+
+            return $cache->clear();
+        } catch (Exception $exception) {
```

Only catch the exceptions you expect. Otherwise you're just hiding errors.